Sources/NIOHTTP1Server/main.swift

```diff
@@ -290,12 +334,23 @@ private final class HTTPHandler: ChannelInboundHandler {
                 ctx.channel.close(promise: nil)
             }
         case .end(_):
-            ()
+            self.state.requestComplete()
         default:
             fatalError("oh noes: \(request)")
         }
     }
 
+    @discardableResult
+    private func completeResponse(_ ctx: ChannelHandlerContext, trailers: HTTPHeaders?) -> EventLoopFuture<Void> {
```

Comment:
@Lukasa I wonder if we should allow to pass in `EventLoopPromise?` and just not return the future here. This would allow to optimize the cases where we not care about there result and keep alive is used.